Firestore/core/test/unit/util/byte_stream_test.cc

```diff
@@ -111,6 +184,16 @@ TEST_P(ByteStreamTest, ReadsNullCharacter) {
   EXPECT_TRUE(result.eof());
 }
 
+TEST_P(ByteStreamTest, ReadsFullStringWithNullCharacter) {
+  // Using explicit string(char*, size) constructor to include \0
+  auto stream =
+      stream_factory_->CreateByteStream(std::string("10{conten\0t}5{\0}", 16));
```

Comment:
Nit: please create a variable for this string and reuse it in the `EXPECT_EQ` below.

Reply:
Done.